test/expect.js

```diff
@@ -790,34 +791,17 @@ describe('expect', function () {
 
     err(function(){
       expect('asd').to.have.property('foo');
-    }, "expected 'asd' to have a property 'foo'");
+    }, "expected 'asd' to have property 'foo'");
     err(function(){
       expect({ foo: { bar: 'baz' } })
         .to.have.property('foo.bar');
-    }, "expected { foo: { bar: 'baz' } } to have a property 'foo.bar'");
-  });
-
-  it('nested.property(name)', function(){
-    expect({ 'foo.bar': 'baz'})
-      .to.not.have.nested.property('foo.bar');
-    expect({ foo: { bar: 'baz' } })
-      .to.have.nested.property('foo.bar');
-
-    expect({ 'foo': [1, 2, 3] })
-      .to.have.nested.property('foo[1]');
-
-    expect({ 'foo.bar[]': 'baz'})
-      .to.have.nested.property('foo\\.bar\\[\\]');
-
-    err(function(){
-      expect({ 'foo.bar': 'baz' })
-        .to.have.nested.property('foo.bar');
-    }, "expected { 'foo.bar': 'baz' } to have a nested property 'foo.bar'");
+    }, "expected { foo: { bar: 'baz' } } to have property 'foo.bar'");
   });
 
   it('property(name, val)', function(){
```

Comment:
IMO it would be good to have tests to see if we're really unable to use both `own` and `nested` together.